pkg/analysis/analysis.go

```diff
@@ -16,11 +16,19 @@ var (
 )
 
 type Pass struct {
-	AnalyzerName  string
-	RootDir       string
-	SourceCodeDir string
-	ResultOf      map[*Analyzer]interface{}
-	Report        func(string, Diagnostic)
+	AnalyzerName string
+	RootDir      string
+	CheckParams  *CheckParams
```

Comment:
Can/does it make sense for `CheckParams` to be `nil`? I don't think it would make sense, in that case I would change this from a `*CheckParams` (pointer) to a `CheckParams` (value). This way it cannot be nil and it cannot panic if trying to access its attributes